code/datums/elements/cleaning.dm

```diff
@@ -40,3 +39,37 @@
 				SEND_SIGNAL(cleaned_human, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRENGTH_BLOOD)
 				cleaned_human.regenerate_icons()
 				to_chat(cleaned_human, "<span class='danger'>[AM] cleans your face!</span>")
+
+/datum/action/cleaning_toggle
+	name = "Floor Cleaning Toggle"
+	desc = "Toggles the automatic floor cleaning"
+	icon_icon = 'icons/obj/vehicles.dmi'
+	button_icon_state = "upgrade"
+	var/toggled = TRUE
+	var/atom/movable/toggle_target = null
```

Comment:
This variable isn't getting Garbage collected when the datum is destroyed

Reply:
Resolved